test/utils/mongoose-connect.js

```diff
@@ -7,7 +7,7 @@ function mongooseConnect() {
   return new P((resolve, reject) => {
     mongoose.connect(
       'mongodb://localhost:27017/forest-test',
-      { useNewUrlParser: true, useUnifiedTopology: true },
+      { useNewUrlParser: true, useUnifiedTopology: true, useFindAndModify: false },
```

Comment:
[MISC] Maybe a dummy question, but are we not supposed to do something about it instead of muting the warnings (I mean warnings probably exist for a good reason)

Reply:
Plus I'm not quite sure that this is a blocking issue for the scope related refactor

Reply:
Indeed, it is absolutely unrelated to the scopes.
So are all the others PRs that i did for now, I'm just cleaning up the source tree before making big changes.

I think that mongoose deprecated their implementation because it can cause race-conditions, while the database version is atomic.

Support for database `findAndModify` was implemented in Mongoose in 2018, more than 3 years ago.
https://github.com/Automattic/mongoose/blob/master/History.md#507--2018-02-23

The configuration options does not mute the warning, it tells mongoose to stop using its own implementation of findAndModify and use the database one instead.

You can see the code in mongoose here:
https://github.com/Automattic/mongoose/blob/312df3d618f6362fe37b5b29b56744b16f22e2ca/lib/query.js#L3669-L3722

We could also replace all calls in the source code.
Searching for `find.*AndUpdate` in the repository yields only 5 matches...

Reply:
> [MISC] (I mean warnings probably exist for a good reason)

I think that mongoose deprecated their implementation because it can cause race-conditions, while the database version is atomic.

Support for database `findAndModify` was implemented in Mongoose in 2018, more than 3 years ago.
https://github.com/Automattic/mongoose/blob/master/History.md#507--2018-02-23

The deprecation notice is 2 years old
https://github.com/Automattic/mongoose/blob/master/History.md#553--2019-04-22

Reply:
> [MISC] Maybe a dummy question, but are we not supposed to do something about it instead of muting the warnings

The configuration options does not mute the warning, it tells mongoose to stop using its own implementation of findAndModify and use the database one instead.

You can see the code in mongoose here:
https://github.com/Automattic/mongoose/blob/312df3d618f6362fe37b5b29b56744b16f22e2ca/lib/query.js#L3669-L3722

We could also replace all calls in the source code.
Searching for `find.*AndUpdate` in the repository yields only 5 matches...

Reply:
> Plus I'm not quite sure that this is a blocking issue for the scope related refactor

Indeed, it is absolutely unrelated to the scopes.
So are all the others PRs that i did for now, I'm just cleaning up the source tree before making big changes.

Reply:
Ok, good for me then, I thought you were unnecessarily muting warning for an unrelated subject 👍